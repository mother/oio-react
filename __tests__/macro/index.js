import fs from 'fs'
import path from 'path'
import pluginTester from 'babel-plugin-tester'
import plugin from 'babel-plugin-macros'

pluginTester({
   plugin,
   babelOptions: { filename: __filename },
   snapshot: true,
   tests: [
      {
         title: '#1 - works with template literals that have no expressions',
         code: fs.readFileSync(`${path.resolve(__dirname)}/fixtures/1.js`, 'utf8')
      },
      {
         title: '#2 - works with tagged template literals with mixed quasis and expressions',
         code: fs.readFileSync(`${path.resolve(__dirname)}/fixtures/2.js`, 'utf8')
      },
      {
         title: '#3 - works with tagged template literals with mixed only expressions',
         code: fs.readFileSync(`${path.resolve(__dirname)}/fixtures/3.js`, 'utf8')
      }
   ]
})
