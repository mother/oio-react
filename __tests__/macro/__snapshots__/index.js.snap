// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`macros #1 - works with template literals that have no expressions: #1 - works with template literals that have no expressions 1`] = `
"
/* eslint-disable no-unused-vars,no-undef,import/newline-after-import */
import OIOResponsiveStringMacro from '../../macro'
const result = OIOResponsiveStringMacro\`100[a] 200[b]\`

      ↓ ↓ ↓ ↓ ↓ ↓

/* eslint-disable no-unused-vars,no-undef,import/newline-after-import */
const result = {
  breakpointsWereSet: true,
  a: \\"100\\",
  b: \\"200\\"
};
"
`;

exports[`macros #2 - works with tagged template literals with mixed quasis and expressions: #2 - works with tagged template literals with mixed quasis and expressions 1`] = `
"
/* eslint-disable no-unused-vars,no-undef,import/newline-after-import */
import OIOResponsiveStringMacro from '../../macro'
const result = OIOResponsiveStringMacro\`200 \${size} \${unit}px [ab] 100px [c]\`

      ↓ ↓ ↓ ↓ ↓ ↓

/* eslint-disable no-unused-vars,no-undef,import/newline-after-import */
const result = {
  a: \`200 \${size} \${unit}px \`,
  b: \`200 \${size} \${unit}px \`,
  c: \` 100px \`
};
"
`;

exports[`macros #3 - works with tagged template literals with mixed only expressions: #3 - works with tagged template literals with mixed only expressions 1`] = `
"
/* eslint-disable no-unused-vars,no-undef,import/newline-after-import */
import OIOResponsiveStringMacro from '../../macro'
const result = OIOResponsiveStringMacro\`\${a}[a]\${b}[b]\${c}\${d}\${e}[cde]\`

      ↓ ↓ ↓ ↓ ↓ ↓

/* eslint-disable no-unused-vars,no-undef,import/newline-after-import */
const result = {
  a: \`\${a}\`,
  b: \`\${b}\`,
  c: \`\${c}\${d}\${e}\`,
  d: \`\${c}\${d}\${e}\`,
  e: \`\${c}\${d}\${e}\`
};
"
`;
